evergreen/run-smoketests.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -o errexit  # Exit the script with error if any of the commands fail
+
+MONGODB_URI=${MONGODB_URI:=mongodb://localhost:27017/}
+VERSION_TAG=$(git describe --tags --abbrev=0 --match "v*")
+PACKAGE_VERSION=${VERSION_TAG:1}
+
+ODATA_PROJECT="./src/MongoDB.AspNetCore.OData/MongoDB.AspNetCore.OData.csproj"
+ODATA_SAMPLE_PROJECT="./tests/MongoDB.AspNetCore.OData.Sample.WebApi/MongoDB.AspNetCore.OData.Sample.WebApi.csproj"
+ODATA_TESTS_PROJECT="./tests/MongoDB.AspNetCore.OData.Tests/MongoDB.AspNetCore.OData.Tests.csproj"
+
+echo Retargeting API tests to use generated package instead of project dependency...
+dotnet clean "./MongoDB.AspNetCore.OData.sln"
+
+dotnet nuget add source "./build/nuget" -n local --configfile "./nuget.config"
+dotnet remove "$ODATA_SAMPLE_PROJECT" reference "$ODATA_PROJECT"
```

Comment:
Just thinking, if we are setting env variable for driver version, then theoretically no need to add and remove packages, can just add a switch in the project.

Reply:
Yep, but the project itself has project reference for debug purposes and testing "current" version. But in smoke tests, we want to use package instead of project reference.

Reply:
Yes, we can still do this conditionally.

Reply:
Discussed, will do after the RC package creation